tests/scenarios/delegated-grant.spec.ts

```diff
@@ -0,0 +1,422 @@
+      expect(participantRoleReply.status.code).to.equal(202);
+
+      // Bob writes a chat message in the thread
+      const chatRecord = await TestDataGenerator.generateRecordsWrite({
+        author       : bob,
+        protocol     : protocolDefinition.protocol,
+        protocolPath : 'thread/chat',
+        contextId    : threadRecord.message.contextId,
+        parentId     : threadRecord.message.recordId,
+      });
+      const chatRecordReply = await dwn.processMessage(bob.did, chatRecord.message, chatRecord.dataStream);
+      expect(chatRecordReply.status.code).to.equal(202);
+
+      // Alice creates a delegated grant for device X to act as Alice.
+      const scope: PermissionScope = {
+        interface : DwnInterfaceName.Records,
+        method    : DwnMethodName.Write,
+        protocol
+      };
+
+      const grantToDeviceX = await PermissionsGrant.create({
+        delegated   : true, // this is a delegated grant
+        dateExpires : Time.createOffsetTimestamp({ seconds: 100 }),
+        description : 'Allow device X to write as me in chat protocol',
+        grantedBy   : alice.did,
+        grantedTo   : bob.did,
+        grantedFor  : alice.did,
+        scope       : scope,
+        signer      : Jws.createSigner(alice)
+      });
+
+      // sanity verify Bob himself is able to read the chat thread from Bob's DWN
+      const recordsQueryByBob = await TestDataGenerator.generateRecordsQuery({
+        author : bob,
+        filter : { protocol }
+      });
+      const bobRecordsQueryReply = await dwn.processMessage(bob.did, recordsQueryByBob.message);
+      expect(bobRecordsQueryReply.status.code).to.equal(200);
+      expect(bobRecordsQueryReply.entries?.length).to.equal(3);
+
+      // sanity verify Alice herself is able to read the chat message from Bob's DWN
+      const recordsQueryByAlice = await RecordsQuery.create({
+        signer       : Jws.createSigner(alice),
+        protocolRole : 'thread/participant',
+        filter       : {
+          protocol,
+          contextId    : threadRecord.message.contextId,
+          protocolPath : 'thread/chat'
+        }
+      });
+      const aliceRecordsQueryReply = await dwn.processMessage(bob.did, recordsQueryByAlice.message);
+      expect(aliceRecordsQueryReply.status.code).to.equal(200);
+      expect(aliceRecordsQueryReply.entries?.length).to.equal(1);
+
+      // verify device X is able to read the chat message from Bob's DWN
+      const recordsQueryByDeviceX = await RecordsQuery.create({
+        signer         : Jws.createSigner(deviceX),
+        delegatedGrant : grantToDeviceX.asDelegatedGrant(),
+        protocolRole   : 'thread/participant',
+        filter         : {
+          protocol,
+          contextId    : threadRecord.message.contextId,
+          protocolPath : 'thread/chat'
+        }
+      });
+      const deviceXRecordsQueryReply = await dwn.processMessage(bob.did, recordsQueryByDeviceX.message);
```

Comment:
might be worth testing that `recordsQueryByDeviceX.recordsQuery.author === alice.did`.

Reply:
Is the new test added what you are looking for or something else?

Reply:
Yep, that's sufficient